pkg/api/v20191231preview/openshiftcluster_validatestatic.go

```diff
@@ -111,13 +112,12 @@ func (sv *openShiftClusterStaticValidator) validateProperties(path string, p *Pr
 	return nil
 }
 
-func (sv *openShiftClusterStaticValidator) validateClusterProfile(path string, cp *ClusterProfile) error {
+func (sv *openShiftClusterStaticValidator) validateClusterProfile(path string, cp *ClusterProfile, isCreate bool) error {
 	if !validate.RxDomainName.MatchString(cp.Domain) {
 		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path+".domain", "The provided domain '%s' is invalid.", cp.Domain)
 	}
-	switch cp.Version {
-	case "", "4.3.0":
-	default:
+	if isCreate && cp.Version != version.OpenShiftVersion ||
+		!isCreate && !validate.RxOpenShiftVersion.MatchString(cp.Version) {
```

Comment:
Disallow empty version.  On create the version must match what we can provision (if the end user doesn't provide the value, we default it in).  On !create, match "4.3.number".